common/esxi_get_guest_config_options.yml

```diff
@@ -154,135 +146,142 @@
-                    })
-           }}
-      when:
-        - guest_supported_disk_ctrls is defined
-        - guest_supported_disk_ctrls.matches is defined
-        - guest_supported_disk_ctrls.matches | length > 0
+        - name: "Set default video RAM size in KB for guest id {{ guest_id }}"
+          ansible.builtin.set_fact:
+            guest_config_options: "{{ guest_config_options | combine({'rec_vram_kb': guest_vram_size.matches[0].defaultValue}) }}"
+          when:
+            - guest_vram_size is defined
+            - guest_vram_size.matches is defined
+            - guest_vram_size.matches | length > 0
 
-    - name: "Get supported disk controllers for guest id {{ guest_id }}"
-      community.general.xml:
-        path: "{{ tmp_config_option_file }}"
-        xpath: "{{ guest_config_options_xpath }}/supportedEthernetCard/e"
-        content: text
-      register: guest_supported_ethernet_card
-      ignore_errors: true
+        - name: "Get supported disk controllers for guest id {{ guest_id }}"
+          community.general.xml:
+            path: "{{ tmp_config_option_file }}"
+            xpath: "{{ guest_config_options_xpath }}/supportedDiskControllerList/e"
+            content: text
+          register: guest_supported_disk_ctrls
+          ignore_errors: true
 
-    - name: "Set supported network adapters for guest id {{ guest_id }}"
-      ansible.builtin.set_fact:
-        guest_config_options: >-
-          {{
-            guest_config_options
-            | combine({
-                 'support_ethernet_card':
-                   guest_supported_ethernet_card.matches | map(attribute='e') | map('extract', vm_device_types) | list
-                      })
-           }}
-      when:
-        - guest_supported_ethernet_card is defined
-        - guest_supported_ethernet_card.matches is defined
-        - guest_supported_ethernet_card.matches | length > 0
+        - name: "Set supported disk controllers for guest id {{ guest_id }}"
+          ansible.builtin.set_fact:
+            guest_config_options: >-
+              {{
+                guest_config_options
+                | combine({
+                     'support_disk_controller':
+                       guest_supported_disk_ctrls.matches | map(attribute='e') | map('extract', vm_device_types) | list
+                        })
+               }}
+          when:
+            - guest_supported_disk_ctrls is defined
+            - guest_supported_disk_ctrls.matches is defined
+            - guest_supported_disk_ctrls.matches | length > 0
 
-    - name: "Get supported USB controllers for guest id {{ guest_id }}"
-      community.general.xml:
-        path: "{{ tmp_config_option_file }}"
-        xpath: "{{ guest_config_options_xpath }}/supportedUSBControllerList/e"
-        content: text
-      register: guest_supported_usb_ctrls
-      ignore_errors: true
+        - name: "Get supported disk controllers for guest id {{ guest_id }}"
```

Comment:
It should be "Get supported newtwork adapters"

Reply:
Nice catch. It has been fixe.